### DISCUSSION TITLE
New "NavSaving" Event

### DISCUSSION THREAD

**Discussion by daerentis** [statamic/cms#10602] [Polls]
We currently have quite a few Events like `CollectionSaving`, `AssetSaving` or `EntrySaving` that fires before actually saving.
Any way this could be implemented? I dont really have the skills to implement it myself.

**Comment by duncanmcclean** [comment]
What's your use case? Being able to see if a nav's settings are updated or being able to see when items are added/updated/removed from the nav?

**Reply by daerentis** [reply]
I need to automatically assign all available sites/locales to the nav collection.
Currently i am doing the same with all collections and the CollectionSaving event. If i use the NavSaved event i get into a infinite loop becaust the re-saving will end up triggering the event again :-)

```php
namespace App\Listeners;

use Statamic\Events\CollectionSaving;
use Statamic\Facades\Site;

class AssignSitesToCollection
{
    public function handle(CollectionSaving $event): void
    {
        $event->collection->sites(Site::all()->pluck('handle')->toArray());
    }
}
```

**Comment by duncanmcclean** [comment]
Added them in a pull request: #10604
